components/camera/camera.go

```diff
@@ -280,143 +206,6 @@ func NewPropertiesError(cameraIdentifier string) error {
-			}
-			var cameraModel transform.PinholeCameraModel
-			cameraModel.PinholeCameraIntrinsics = props.IntrinsicParams
-
-			if props.DistortionParams != nil {
-				cameraModel.Distortion = props.DistortionParams
-			}
-
-			actualSystem = &cameraModel
-		}
-	}
-	return &videoSource{
-		system:       actualSystem,
-		videoSource:  source,
-		videoStream:  gostream.NewEmbeddedVideoStream(source),
-		actualSource: source,
-		imageType:    imageType,
-	}, nil
-}
-
-// videoSource implements a Camera with a gostream.VideoSource.
-type videoSource struct {
-	rtpPassthroughSource rtppassthrough.Source
-	videoSource          gostream.VideoSource
-	videoStream          gostream.VideoStream
-	actualSource         interface{}
-	system               *transform.PinholeCameraModel
-	imageType            ImageType
-}
-
-func (vs *videoSource) Stream(ctx context.Context, errHandlers ...gostream.ErrorHandler) (gostream.VideoStream, error) {
-	return vs.videoSource.Stream(ctx, errHandlers...)
-}
-
-// Images is for getting simultaneous images from different sensors
-// If the underlying source did not specify an Images function, a default is applied.
-// The default returns a list of 1 image from ReadImage, and the current time.
-func (vs *videoSource) Images(ctx context.Context) ([]NamedImage, resource.ResponseMetadata, error) {
-	ctx, span := trace.StartSpan(ctx, "camera::videoSource::Images")
-	defer span.End()
-	if c, ok := vs.actualSource.(ImagesSource); ok {
-		return c.Images(ctx)
-	}
-	img, release, err := ReadImage(ctx, vs.videoSource)
-	if err != nil {
-		return nil, resource.ResponseMetadata{}, errors.Wrap(err, "videoSource: call to get Images failed")
-	}
-	defer func() {
-		if release != nil {
-			release()
-		}
-	}()
-	ts := time.Now()
-	return []NamedImage{{img, ""}}, resource.ResponseMetadata{CapturedAt: ts}, nil
-}
-
-// NextPointCloud returns the next PointCloud from the camera, or will error if not supported.
-func (vs *videoSource) NextPointCloud(ctx context.Context) (pointcloud.PointCloud, error) {
-	ctx, span := trace.StartSpan(ctx, "camera::videoSource::NextPointCloud")
-	defer span.End()
-	if c, ok := vs.actualSource.(PointCloudSource); ok {
-		return c.NextPointCloud(ctx)
-	}
-	if vs.system == nil || vs.system.PinholeCameraIntrinsics == nil {
-		return nil, transform.NewNoIntrinsicsError("cannot do a projection to a point cloud")
-	}
-	img, release, err := vs.videoStream.Next(ctx)
-	defer release()
-	if err != nil {
-		return nil, err
-	}
-	dm, err := rimage.ConvertImageToDepthMap(ctx, img)
-	if err != nil {
-		return nil, errors.Wrapf(err, "cannot project to a point cloud")
-	}
-	return depthadapter.ToPointCloud(dm, vs.system.PinholeCameraIntrinsics), nil
-}
-
-func (vs *videoSource) Projector(ctx context.Context) (transform.Projector, error) {
-	if vs.system == nil || vs.system.PinholeCameraIntrinsics == nil {
-		return nil, transform.NewNoIntrinsicsError("No features in config")
-	}
-	return vs.system.PinholeCameraIntrinsics, nil
-}
```

Comment:
@nicksanford @bhaney so it doesn't get lost in the move - this was completely removed in favour of the helper function now in rimage.

Reply:
So this is an anticipated merge conflict?

Reply:
This was only used once in the entirety of rdk in the code in `rimage`. Once i delete it, if the method is added back, since I've removed projectorProvider, it won't be part of the camera interface but will just be dead unless you specifically import `videosource` from the camera package or the `transform` or `webcam "`videosource`" go packages to call it. Hopefully no merge conflicts.